### PR TITLE
External Audit Storage: add usage event

### PR DESCRIPTION
**Pull request by nklaassen**
This adds a regular usage event every time credentials are refreshed for the External Audit Storage feature.

It's desirable to regularly submit an event instead of just a one-off when the feature is turned on, because there's no guarantee an individual event will actually make it to prehog/posthog and won't be lost. This event should be emitted roughly every 45 minutes.

Cloud PR: https://github.com/gravitational/cloud/pull/6772

### CODE REVIEW

File: lib/usagereporter/teleport/types.go
Comment (espadolini) [inline lib/usagereporter/teleport/types.go:905]: If we add fields to `prehogv1a.ExternalAuditStorageAuthenticateEvent` this will end up sending them straight through without any anonymization.
```suggestion
			ExternalAuditStorageAuthenticate: &prehogv1a.ExternalAuditStorageAuthenticateEvent{},
```

Technically this would be equivalent, but it might raise some eyebrows.
```suggestion
			ExternalAuditStorageAuthenticate: nil,
```

Comment (nklaassen) [reply lib/usagereporter/teleport/types.go:905]: Good point!

**Review by espadolini — APPROVED** [review]

**Review by AntonAM — APPROVED** [review]

**Comment by espadolini** [comment]
Added `do-not-merge` while waiting for the cloud PR to get merged first, feel free to remove it once that one is merged.

**Pull request merged**
